commit d651abc097d91fac57f33b5f9447d0a9183f58e7	d651abc097d91fac57f33b5f9447d0a9183f58e7
Author: erouault <erouault>
Date:   Wed Nov 9 23:00:49 2016 +0000

    * libtiff/tif_strip.c: make TIFFNumberOfStrips() return the td->td_nstrips
    value when it is non-zero, instead of recomputing it. This is needed in
    TIFF_STRIPCHOP mode where td_nstrips is modified. Fixes a read outsize of
    array in tiffsplit (or other utilities using TIFFNumberOfStrips()).
    Fixes http://bugzilla.maptools.org/show_bug.cgi?id=2587

diff --git a/ChangeLog b/ChangeLog
index 48fb75d3..4ff52810 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,11 @@
+2016-11-10 Even Rouault <even.rouault at spatialys.com>
+
+	* libtiff/tif_strip.c: make TIFFNumberOfStrips() return the td->td_nstrips
+	value when it is non-zero, instead of recomputing it. This is needed in
+	TIFF_STRIPCHOP mode where td_nstrips is modified. Fixes a read outsize of
+	array in tiffsplit (or other utilities using TIFFNumberOfStrips()).
+	Fixes http://bugzilla.maptools.org/show_bug.cgi?id=2587
+
 2016-11-04 Even Rouault <even.rouault at spatialys.com>
 
 	* libtiff/tif_predic.c: fix memory leaks in error code paths added in
diff --git a/libtiff/tif_strip.c b/libtiff/tif_strip.c
index 1676e47d..4c46ecf5 100644
--- a/libtiff/tif_strip.c
+++ b/libtiff/tif_strip.c
@@ -63,6 +63,15 @@ TIFFNumberOfStrips(TIFF* tif)
 	TIFFDirectory *td = &tif->tif_dir;
 	uint32 nstrips;
 
+    /* If the value was already computed and store in td_nstrips, then return it,
+       since ChopUpSingleUncompressedStrip might have altered and resized the
+       since the td_stripbytecount and td_stripoffset arrays to the new value
+       after the initial affectation of td_nstrips = TIFFNumberOfStrips() in
+       tif_dirread.c ~line 3612.
+       See http://bugzilla.maptools.org/show_bug.cgi?id=2587 */
+    if( td->td_nstrips )
+        return td->td_nstrips;
+
 	nstrips = (td->td_rowsperstrip == (uint32) -1 ? 1 :
 	     TIFFhowmany_32(td->td_imagelength, td->td_rowsperstrip));
 	if (td->td_planarconfig == PLANARCONFIG_SEPARATE)
